src/Tribe/Integrations/WPML/Meta.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+
+class Tribe__Events__Integrations__WPML__Meta {
+
+	/**
+	 * @var Tribe__Events__Integrations__WPML__Meta
+	 */
+	protected static $instance;
+
+	/**
+	 * @return Tribe__Events__Integrations__WPML__Meta
+	 */
+	public static function instance() {
+		if ( empty( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Translates post id in the Event meta data.
+	 *
+	 * @param string $value
+	 * @param int    $object_id
+	 * @param string $meta_key
+	 *
+	 * @return string The translated id for _EventOrganizerID & _EventVenueID.
+	 */
+	public function translate_post_id( $value, $object_id, $meta_key ) {
+		global $wpdb;
+
```

Comment:
Seems like the only thing on this function that changes is the Post Type. So I think adding: 
- An early return
- And just condition the type 

Might be a little bit more compact, let me know what you think about this @dgwatkins 

```php
if ( '_EventOrganizerID' !== $meta_key || '_EventVenueID' !== $meta_key ) {
	return $value;
}

$value = $wpdb->get_var( $wpdb->prepare(
	"SELECT meta_value FROM $wpdb->postmeta WHERE post_id = %d AND meta_key = %s",
	$object_id,
	$meta_key
) );

$type = '_EventOrganizerID' === $meta_key 
	? Tribe__Events__Organizer::POSTTYPE
	: Tribe__Events__Venue::POSTTYPE;

$value = apply_filters( 'wpml_object_id', $value, $type, true );
```

Reply:
I updated the code, indeed its more compact.

Reply:
Great thanks for your contribution.